ic-stable-structures/src/structure/cache/btreemap.rs

```diff
@@ -0,0 +1,194 @@
+use std::{cell::RefCell, hash::Hash};
+
+use crate::structure::*;
+use ic_exports::stable_structures::{memory_manager::MemoryId, BoundedStorable};
+use mini_moka::unsync::{Cache, CacheBuilder};
+
+/// A LRU Cache for StableBTreeMap
+pub struct CachedStableBTreeMap<K, V>
+where
+    K: BoundedStorable + Clone + Hash + Eq + PartialEq + Ord,
+    V: BoundedStorable + Clone,
+{
+    inner: StableBTreeMap<K, V>,
```

Comment:
It would be logical to have the inner structure be generic implementing `BTreeMapStructure<K, V>`

Reply:
If the map parameter is the first in the list of generic parameters the following code works:

```
let cache = CachedStableBTreeMap< StableBTreeMap<u64, u64>, _, _> = CachedStableBTreeMap::new(StableBTreeMap::new(...), max_cache_items);

```

which is slightly better, but anyway I agree that it is not that important